validator/vars.go

```diff
@@ -42,20 +42,23 @@ func VariableValues(schema *ast.Schema, op *ast.OperationDefinition, variables m
 		}
 
 		if hasValue {
```

Comment:
Is this condition even required?  Seems like the previous block either returns or sets `hasValue` to `true`

Reply:
yeah, its awkward code because it tries to read close to the spec, but there is a live path that falls through when the default is set.